readthedocs/proxito/tests/test_middleware.py

```diff
@@ -73,6 +73,42 @@ def test_canonical_cname_redirect(self):
             self.assertTrue(hasattr(request, 'canonicalize'))
             self.assertEqual(request.canonicalize, 'canonical-cname')
 
+    def test_subproject_redirect(self):
+        """Requests to a subproject should redirect to the domain of the main project."""
+        subproject = get(
+            Project,
+            name='subproject',
+            slug='subproject',
+            users=[self.owner],
+            privacy_level=PUBLIC,
+        )
+        subproject.versions.update(privacy_level=PUBLIC)
+        get(
+            ProjectRelationship,
+            parent=self.pip,
+            child=subproject,
+        )
+
+        for url in (self.url, '/subdir/', '/en/latest/'):
+            request = self.request(url, HTTP_HOST='subproject.dev.readthedocs.io')
+            res = self.run_middleware(request)
+            self.assertIsNone(res)
+            self.assertEqual(getattr(request, 'canonicalize', None), 'subproject-main-domain')
+
+        # Using a custom domain in a subproject isn't supported (or shouldn't be!).
```

Comment:
Added a test for this to check isn't failing since it could happen, but we can also move the subdomain check first. Think we have talked about checking this case in the domain creation form.

Reply:
Yea, we shouldn't allow this.